src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,190 @@
+import os
+import re
+import logging
+from common.requester import DelayedRequester
+from common.storage.image import ImageStore
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 35
+DELAY = 5.0
+RETRIES = 3
+PROVIDER = "brooklynmuseum"
+ENDPOINT = "https://www.brooklynmuseum.org/api/v2/object/"
+API_KEY = os.getenv("BROOKLYN_MUSEUM_API_KEY", "nokeyprovided")
+
+delay_request = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+HEADERS = {
+    "api_key": API_KEY
+}
+
+DEFAULT_QUERY_PARAM = {
+    "has_images": 1,
+    "rights_type_permissive": 1,
+    "limit": LIMIT,
+    "offset": 0
+}
+
+
+def main():
+    logger.info("Begin: Brooklyn museum provider script")
+    offset = 0
+    condition = True
+    while condition:
+        query_param = _get_query_param(offset)
+        objects_batch = _get_object_json(query_param=query_param)
+        logger.debug(len(objects_batch))
+        if len(objects_batch) > 0:
+            image_count = _process_objects_batch(objects_batch)
+            logger.debug(f"Images till now {image_count}")
+            offset += LIMIT
+        else:
+            condition = False
+    total_images = image_store.commit()
+    logger.info(f"Total images recieved {total_images}")
+
+
+def _get_query_param(offset=0,
+                     default_query_param=DEFAULT_QUERY_PARAM):
+    query_param = default_query_param.copy()
+    query_param.update(
+        offset=offset
+    )
+    return query_param
+
+
+def _get_object_json(headers=HEADERS,
+                     endpoint=ENDPOINT,
+                     retries=RETRIES,
+                     query_param=None):
+    response_json = None
+    for tries in range(retries):
+        try:
+            response = delay_request.get(
+                endpoint,
+                query_param,
+                headers=headers
+            )
+            data = response.json()
+            if data and data.get("message", "").lower() == "success.":
+                response_json = data.get("data", "")
+                break
+        except Exception as e:
+            logger.warning(f"response not captured due to {e}")
+    return response_json
```

Comment:
You can skip setting the `Nonetype` default, and just return response_json if you get one.  Also, the `get` calls are extraneous, since the whole thing is wrapped in a `try`, and you *want* to fail if something is missing.
```suggestion
    for tries in range(retries):
        try:
            response = delay_request.get(
                endpoint,
                query_param,
                headers=headers
            )
            data = response.json()
            assert data["message"].lower() == "success."
            return data["data"]
        except Exception as e:
            logger.warning(f"response not captured due to {e}")
```

Reply:
@mathemancer how about this ? 
we can avoid the ``try`` since the logger warnings are handled in ```requester module```. 
```
def _get_object_json(headers=HEADERS,
                     endpoint=ENDPOINT,
                     retries=RETRIES,
                     query_param=None):
   for tries in range(retries):
	response = delay_request.get(
                endpoint,
                query_param,
                headers=headers
            )
        data = response.json()
        if data and data.get("message", "").lower() == "success.":
        	response_json = data.get("data", "")
                break
	else:
		response_json = None
   return response_json

```

Reply:
Sure, this also works.  The default for `data.get("data", ...)` should be the default `None` in this case, no?  I.e., if we can't find the `response_json` we want, shouldn't we return `None` instead of an empty string?

Reply:
The default for the `dict.get` method, that is

Reply:
yes that needs to made to ```None```.